File: stack&queue/leetcode_1021/solution.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        stack = []
        balance = 0

        for char in s:
            if char == '(':
                if balance > 0:
                    stack.append(char)
                
                balance += 1
            else:
                balance -= 1
                
                if balance > 0:
                    stack.append(char)
            
        return ''.join(stack)
        

File: stack&queue/leetcode_1021/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    const stack = [];
    let balance = 0;

    for(const char of s) {
        if (char === '(') {
            if(balance > 0) {
                stack.push(char);
            }
            
            balance ++;
        } else {
            balance --;
            
            if(balance > 0) {
                stack.push(char);
            }
        }
    }

    return stack.join('')
};